src/htm/algorithms/SpatialPooler.hpp

```diff
@@ -109,13 +117,14 @@ class SpatialPooler : public Serializable
         column will have a max square potential pool with sides of
         length (2 * potentialRadius + 1).
 
-  @param potentialPct The percent of the inputs, within a column's
-        potential radius, that a column can be connected to. If set to
-        1, the column will be connected to every input within its
+  @param potentialPct The percent of the inputs, within a column's //TODO make this "automated" depending on #potentialRadius & numColumns. 
```

Comment:
- rename to `columnInputOverlapPct`
- remove and make it a function of `Fn(#columns, input area, potential radius,local area pct, *prefer-local-vs-global) `
  - #columns + -> Fn -
  - area + -> Fn -
  - pot radius + -> Fn +
  - local area Pct + -> Fn +
  - prefer local + -> Fn +

- the Fn represents "prefer local, details" (over global, holistic)
- new smart param "prefer local" 0..1.0

Reply:
this works well, see #533  for successful demonstration